File: Pset 2/Caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc == 2 )
    {
        int local = 0;
        for (int i = 0, n = strlen(argv[1]); i < n; i++)
        {
            if (!(isdigit(argv[1][i])))
            {
                local = 1;
                break;
            }
        }
        if (local == 0)
        {
            printf("Success\n%s\n", argv[1]);
        }
        else
        {
            printf("Usage: ./caesar key\n");
            return 1;
        }
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }

    int key = atoi(argv[1]);



    string plain = get_string("plaintext: ");
    printf("\n");


    int numbers[strlen(plain)];

    for (int i = 0, n = strlen(plain); i < n; i++)
    {
        if (islower(plain[i]))      //check if isupper or islower
        {
            numbers[i] = (int) (plain[i] - 'a');        // (int) transforms to ascii        - 'a' ???
        }
        //do something with the non-alpha charachters
        else      //think it's isupper
        {
            numbers[i] = (int) (plain[i] - 'A');
        }
    }

    int new_numbers[strlen(plain)];

    for (int i = 0, n = strlen(plain); i < n; i++)
    {
        new_numbers[i] = (numbers[i] + key)%26;
    }

    //transform "back" everything
    char cipher[strlen(plain)];
    for (int i = 0, n = strlen(plain); i < n; i++)
    {
        if (isalpha(plain[i]))
        {
            if (islower(plain[i]))      //check if isupper or islower
            {
                cipher[i] = (char) (new_numbers[i] + 'a');        // (int) transforms to ascii    - 'a' ???
            }
            //do something with the non-alpha charachters
            else      //think it's isupper
            {
                cipher[i] = (char) (new_numbers[i] + 'A');
            }
        }
        else
        {
            cipher[i] = plain[i];
        }
    }



    //output

    printf("ciphertext: ");
    for (int i = 0, n = strlen(plain); i < n; i++)
    {
        printf("%c", cipher[i]);
    }
    printf("\n");
}

File: Pset 6/caesar.py
import sys
import cs50

if len(sys.argv) == 2:
    if not sys.argv[1].isdigit():
        sys.exit("key must be a digit")
else:
    sys.exit("Usage: caesar key")

key = int(sys.argv[1])

plain = cs50.get_string("plaintext: ")

numbers = []

for c in plain:
    if c.islower():
        numbers.append(ord(c) - ord('a'))
    else:               #dealing with non_alpha chars later
        numbers.append(ord(c) - ord('A'))

new_numbers = []

n = 0
for c in plain:
    new_numbers.append((numbers[n] + key)%26)
    n += 1

#transform "back" everything
cipher = []
m = 0
for c in plain:
    if c.isalpha():
        if c.islower():
            cipher.append(chr(new_numbers[m] + ord('a')))
        else:       #it's upper
            cipher.append(chr(new_numbers[m] + ord('A')))
    else:
        cipher.append(plain[m])
    m += 1

print("ciphertext: ", end="")
k = 0
for c in plain:
    print(f"{cipher[k]}", end="")
    k += 1
print()